dev/PipsPager/PipsPager.cpp

```diff
@@ -516,6 +477,93 @@ void PipsPager::OnNextButtonClicked(const IInspectable& sender, const winrt::Rou
     SelectedPageIndex(SelectedPageIndex() + 1);
 }
 
+void PipsPager::OnGotFocus(const winrt::RoutedEventArgs& args)
+{
+    if (const auto btn = args.OriginalSource().try_as<winrt::Button>())
+    {
+        // If the element inside the Pager is already keyboard focused
+        // and the user will use the mouse to focus on something else
+        // the LostFocus will not be triggered on keyboard focused element
+        // while GotFocus will be triggered on the new mouse focused element.
+        // We account for this scenario and update m_isFocused in case
+        // user will use mouse while being in keyboard focus.
+        if (btn.FocusState() != winrt::FocusState::Pointer)
+        {
+            m_isFocused = true;
+            UpdateNavigationButtonVisualStates();
+        }
+        else
+        {
+            m_isFocused = false;
+        }
+    }
+}
+
+// In order to avoid switching visibility of the navigation buttons while moving focus inside the Pager,
+// wee'll check if the next focused element is inside the Pager.
+void PipsPager::LosingFocus(const IInspectable& sender, const winrt::LosingFocusEventArgs& args)
+{
+    if (const auto repeater = m_pipsPagerRepeater.get())
+    {
+        if (const auto nextFocusElement = args.NewFocusedElement().try_as<winrt::UIElement>())
+        {
+            m_ifNextFocusElementInside = repeater.GetElementIndex(nextFocusElement) != -1
```

Comment:
If the next focusable element will be inside the PipsPager, shouldn't GotFocus or GettingFocus get raised? Not having to use a tracker ref would be ideal since those aren't cheap.

Reply:
The problem is - if we're changing the focus inside the Pager from one element to another, the losing (and then lost focus) will be called, what will trigger update to the visibility of the buttons unless we know for sure that the next element that will be focused is inside the Pager.
GettingFocus or GotFocus don't help us in this scenario because the focus will already be changed. So if we move the focus away from the Pager, we already had to update the visibility in the LostFocus.

Reply:
We can also call `GetTemplateChildT<winrt::Button>(c_previousPageButtonName, *this))` but not sure if it's cheaper to do than keeping a tracker ref :)

Reply:
Hmm I see. This sounds more complicated than it should be but that's just how the focus system works I guess (though it's a bit confusing to me that if the focus moves within a control this is being treated as focus loss and gain).

Reply:
> We can also call GetTemplateChildT<winrt::Button>(c_previousPageButtonName, *this)) but not sure if it's cheaper to do than keeping a tracker ref :)

No visual tree walking is even more expensive. While tracker ref is not ideal it's probably the best in this situation.

Reply:
Yep, that confuses me too:)

Reply:
@chingucoding if you can think of another, simpler of doing this, please, lmk :)

Reply:
> Hmm I see. This sounds more complicated than it should be but that's just how the focus system works I guess (though it's a bit confusing to me that if the focus moves within a control this is being treated as focus loss and gain).

Never mind, I'm an idiot. The focus events bubble up the visual tree. So if the focus moves within the control, one control looses focus, that event bubbles up and as such the LostFocus event gets raised on PipsPager. Similar thing is the case for GettingFocus and GotFocus.

> @chingucoding if you can think of another, simpler way of doing this, please, lmk :)

Another possibility would be to listen to the focus managers events to determine if the focus actually changed to the PipsPager or another control. Though this pattern is discouraged as this can decrease performance (imagine every control would listen to an event fired globally). So this pattern we currently have is pretty good. It might be possible to get a better result with the FocusManager events but not sure if that's a good idea in general.